### ISSUE TITLE
Misc followup from twitter.

### ISSUE THREAD

**Issue by Carreau** [astrojuanlu/sphinx-github-role#16]
1) docs don't seem to have easy link back to github. 
2) Do you think that non-github / gitlab will be a potential extensions, so you can just do `` :issue:`#31` ``  regardless of the vcs, 
and this leads to 
3) Do you believe there could be a project mapping org -> github|gitlab, so that `` :issue:`projongithub#31` `` and `` :issue:`projongitlab#31` ``

4) which is not dependent of previous ones, I'd love to have some standardization and use the same extensions on many projects of the scientific ecosystem, and I can try to replace the one we use in IPython when you belive this is ready enough.

**Comment by astrojuanlu** [comment]
Thanks a lot @Carreau !

1. Great point, will change it.
2. For consistency, `` :gitlab:`#31` `` would be the natural next step, right? And then
3. Without thinking too much about it, your proposal sounds good to have something like `` :issue:`#31` ``, so it could come after (2).
4. Yeah, I think there's too much copy-pasted code in Sphinx projects. I'd love to have these small extensions that do common tasks.

**Comment by Carreau** [comment]
> * For consistency, `` :gitlab:`#31` `` would be the natural next step, right? And then

I guess that also depends wether there is a single URL for gitlab that redirect. For github, helpfully they redirect from /issues/X to pull/X when relevant.  So not sure how easy it is. I thing some extensions may also render the test of ``:pull:`#1` `` and `` :issue:`#1` `` differently, I guess the extension could try to poll github to see if it's issue or PR (or maybe even JS could check if issue/pr is opened at some point ?

And my other reason for havving :issue: or :pull: vs :gh: or :gl: is that then it's uniform across project regardless of their hosting platform :-)

**Comment by astrojuanlu** [comment]
> I guess the extension could try to poll github to see if it's issue or PR (or maybe even JS could check if issue/pr is opened at some point ?

Yep, that's the topic of https://github.com/astrojuanlu/sphinx-github-role/issues/5

> And my other reason for havving :issue: or :pull: vs :gh: or :gl: is that then it's uniform across project regardless of their hosting platform :-)

Mostly agreed - but GitLab users might get offended by using `:pull:` instead of `:merge:` 😛

**Comment by Carreau** [comment]
> Mostly agreed - but GitLab users might get offended by using `:pull:` instead of `:merge:` 😛

:-)
